src/constraint_system.rs

```diff
@@ -222,6 +260,22 @@ impl<'cs, Scalar: PrimeField, CS: ConstraintSystem<Scalar>> ConstraintSystem<Sca
     fn get_root(&mut self) -> &mut Self::Root {
         self.0.get_root()
     }
+
+    fn is_witness_generator(&mut self) -> bool {
+        self.get_root().is_witness_generator()
```

Comment:
How about `self.0.is_witness_generator()`?

Reply:
These express the intention and are more foolproof as written, I think. But yes, I believe your rewrite is equivalent given how namespaces are implemented.

Reply:
Or maybe not. https://github.com/filecoin-project/bellperson/pull/309#discussion_r1215036785